frontend/express/public/javascripts/countly/vue/core.js

```diff
@@ -122,8 +122,27 @@
                     setActiveApp: function(state, activeApp) {
                         state.activeApp = activeApp;
                     },
+                    /**
+                     * 
+                     * @param {*} state 
+                     * @param {*} additionalApps
+                     * Additional app can be Array or an
+                     * individual object
+                     */
                     addToAllApps: function(state, additionalApps) {
-                        state.allApps.push(additionalApps);
+                        if (Array.isArray(additionalApps)) {
+                            var allApps = state.allApps.concat(additionalApps);
+                            state.allApps = allApps;
+                            
+                        } else {
+                            state.allApps.push(additionalApps);
+                        }
+                    },
+                    removeFromAllApps: function(state, appToRemove) {
+                        state.allApps.splice(state.allApps.findIndex(a => a.value === appToRemove.value) , 1)
```

Comment:
Please use indexOf instead of findIndex as its not supported by IE. Also i think grunt will complain about it.
Secondly, i believe this statement will always remove an app from allApps, nomatter if its found or not.
So lets say appToRemove is not found in allApps, returned index will be -1.
and state.allApps.splice(-1,1) will remove the last entry from allApps, which would be wrong.

I suggest you simply use filter here as follows.
`state.allApps = state.allApps.filter(function(a){ return a.value !== appToRemove.value})`

Reply:
one more thing, key name called value is something that is only available in the component.
actual apps object does not have value key.
value is only set [here](https://github.com/Countly/countly-server/blob/ea21f63d7ce1f668cb87539b61634e11b18deffa/frontend/express/public/javascripts/countly/vue/components/sidebar.js#L54) since the component uses it internally.

Therefore, you should use _id here instead of value.